src/main/java/ch/ivyteam/ivy/maven/validate/ValidateMojo.java

```diff
@@ -0,0 +1,67 @@
+package ch.ivyteam.ivy.maven.validate;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+@Mojo(name = ValidateMojo.GOAL, requiresProject = true)
+public class ValidateMojo extends AbstractMojo {
+
+  public static final String GOAL = "validate";
+  protected static final String PLUGIN_GROUPID = "com.axonivy.ivy.ci";
+  protected static final String PLUGIN_ARTIFACTID = "project-build-plugin";
+
+  @Parameter(defaultValue = "${session}", readonly = true, required = true)
+  private MavenSession session;
+
+  @Override
+  public void execute() throws MojoExecutionException, MojoFailureException {
+    validateConsistentPluginVersion(session.getAllProjects());
+  }
+
+  protected void validateConsistentPluginVersion(List<MavenProject> projects) throws MojoExecutionException {
+    Map<String, Set<MavenProject>> versionToProjectsMap = new HashMap<>();
+    for (var project : projects) {
+      for (var plugin : project.getBuild().getPlugins()) {
+        if (PLUGIN_GROUPID.equals(plugin.getGroupId()) && PLUGIN_ARTIFACTID.equals(plugin.getArtifactId())) {
+          var version = plugin.getVersion();
+          // Skip checking plug ins that do not have a version
+          if (version == null) {
+            continue;
+          }
+          getLog().debug(PLUGIN_GROUPID + ":" + plugin.getArtifactId() + ":" + version + " configured in " + project);
+          var projectSet = versionToProjectsMap.get(version);
+          if (projectSet == null) {
+            projectSet = new LinkedHashSet<>();
+            versionToProjectsMap.put(version, projectSet);
+          }
+          projectSet.add(project);
+        }
+      }
+    }
+    if (versionToProjectsMap.size() > 1) {
+      var versions = new ArrayList<>(versionToProjectsMap.keySet());
+      Collections.sort(versions);
+      getLog().error("Several versions of project-build-plugins are configured " + versions + ":");
+      for (var version : versions) {
+        getLog().error(version + ":");
+        for (var project : versionToProjectsMap.get(version)) {
+          getLog().error("\t" + project.toString());
```

Comment:
I'd only state the project "name" rather than it's toSTring representation.
The names are well known from the reactor output ... and we would gain a less verbose output